### DISCUSSION TITLE
[BUG] Review with two teams as CODEOWNER not removing second team

### DISCUSSION THREAD

**Discussion by pbrisbin** [community/community#24010] [New to GitHub]
<p>Sorry if this isn’t the right place to report a bug on GitHub, but I was directed here from <a href="https://github.community/t/submitting-bug-reports-for-github-com/10207">this</a>, and did my best to choose an appropriate Category.</p>
<hr>
<h3>
<a name="scenario-1" class="anchor" href="#scenario-1"></a>Scenario</h3>
<p>Setup <code>CODEOWNERS</code> with two teams for a particular area of the application:</p>
<pre><code class="lang-auto">/sql/ @freckle/ops @freckle/databasers
</code></pre>
<p>Open a PR in that area.</p>
<h3>
<a name="expected-2" class="anchor" href="#expected-2"></a>Expected</h3>
<p>A team member from each team should be automatically assigned, replacing both teams on the PR.</p>
<h3>
<a name="actual-3" class="anchor" href="#actual-3"></a>Actual</h3>
<p>A team member from each team is automatically assigned, but only the first team is replaced. The second team remains on the PR.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/7/1/716ac5fc3a3e20e33724f6a27ea9765827cf9185.png" data-download-href="/uploads/short-url/gbkIkJJqqPfYdeV5HyJBdYiwobr.png?dl=1" title="" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/50812/181080623-61f58261-f1dc-4158-a5f5-4a7f76cfa5cd.png" alt="" data-base62-sha1="gbkIkJJqqPfYdeV5HyJBdYiwobr" width="690" height="102" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/7/1/716ac5fc3a3e20e33724f6a27ea9765827cf9185_2_690x102.png, https://aws1.discourse-cdn.com/github/original/3X/7/1/716ac5fc3a3e20e33724f6a27ea9765827cf9185.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/7/1/716ac5fc3a3e20e33724f6a27ea9765827cf9185.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/7/1/716ac5fc3a3e20e33724f6a27ea9765827cf9185_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename"></span><span class="informations">812×121 18.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Leaving that second team on the PR means all members of that team see the PR as an active request and get notifications on it, rather than just the round-robin-assigned member.</p>
<h3>
<a name="notes-commentary-4" class="anchor" href="#notes-commentary-4"></a>Notes / Commentary</h3>
<p>At first, I thought it was just expected that review from two teams as owners didn’t work, and would only automatically assign from the first listed team. However, the fact that it gets it <em>half</em> right (it assigns a user from each team, but only replaces the first) indicates this is likely a bug.</p>

**Comment by canuckjacq** [comment]
<p>:wave:</p>
<p>Thanks for opening this! Before we look into this further, can I get you to confirm that both teams have exactly the same settings for this feature? There’s a step-by-step here:</p>
<aside class="onebox githubfolder">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/github/setting-up-and-managing-organizations-and-teams/managing-code-review-assignment-for-your-team#configuring-code-review-assignment" target="_blank" rel="noopener">docs.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://docs.github.com/en/github/setting-up-and-managing-organizations-and-teams/managing-code-review-assignment-for-your-team#configuring-code-review-assignment" target="_blank" rel="noopener">Managing code review assignment for your team - GitHub Docs</a></h3>

<p><a href="https://docs.github.com/en/github/setting-up-and-managing-organizations-and-teams/managing-code-review-assignment-for-your-team#configuring-code-review-assignment" target="_blank" rel="noopener">//docs.github.com/en/github/setting-up-and-managing-organizations-and-teams/managing-code-review-assignment-for-your-team</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by pbrisbin** [comment, marked as answer]
<p>Wow good call! You’re totally right that we had our settings wrong:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/f/8/f8d9400d5983796504ed9f6afdcc43a64eb3026b.png" data-download-href="/uploads/short-url/zvpRSLyFhx9UjM4aiYCsVXZVmxJ.png?dl=1" title="" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/50812/181080624-3ab127de-98ce-4410-9063-02a01e82e4a2.png" alt="" data-base62-sha1="zvpRSLyFhx9UjM4aiYCsVXZVmxJ" width="507" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/f/8/f8d9400d5983796504ed9f6afdcc43a64eb3026b_2_507x500.png, https://aws1.discourse-cdn.com/github/optimized/3X/f/8/f8d9400d5983796504ed9f6afdcc43a64eb3026b_2_760x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/f/8/f8d9400d5983796504ed9f6afdcc43a64eb3026b_2_1014x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/f/8/f8d9400d5983796504ed9f6afdcc43a64eb3026b_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename"></span><span class="informations">1033×1018 98.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>That totally explains the behavior, it’s doing exactly what we said to do.</p>
<p>I’ve now fixed it:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/a/1/a183b18fb41ad18f13d525dfcbf57c3c6d614342.png" data-download-href="/uploads/short-url/n2P1y0GyzohURqWspJXXRMZqm4O.png?dl=1" title="" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/50812/181080625-60e08162-3177-4c61-afe7-4198d0da720c.png" alt="" data-base62-sha1="n2P1y0GyzohURqWspJXXRMZqm4O" width="401" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/a/1/a183b18fb41ad18f13d525dfcbf57c3c6d614342_2_401x500.png, https://aws1.discourse-cdn.com/github/optimized/3X/a/1/a183b18fb41ad18f13d525dfcbf57c3c6d614342_2_601x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/a/1/a183b18fb41ad18f13d525dfcbf57c3c6d614342_2_802x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/a/1/a183b18fb41ad18f13d525dfcbf57c3c6d614342_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename"></span><span class="informations">1004×1249 110 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I’m curious, since we have some members who are on both teams, does GitHub attempt to avoid picking the same person in both round-robins?</p>

**Comment by canuckjacq** [comment]
<p>I’m pretty sure the process screens out users who are already named reviewers (which would include the one named in the 1st round robin selection).</p>

**Comment by pbrisbin** [comment]
<p>Sick. I don’t know how (or if I should) “close” a thread, but this is solved.</p>

**Comment by canuckjacq** [comment]
<p>:tada:</p>
<p>You can always select the most helpful response as a “solution” (it’s the :ballot_box_with_check: near the :heart: at the bottom of the reply) - that’ll make it quick for someone else with the same problem to find their fix!</p>
